packages/docgen/src/engine.js

```diff
@@ -1,24 +1,20 @@
 /**
  * External dependencies.
  */
-const espree = require( 'espree' );
+const babel = require( '@babel/parser' );
 const { flatten } = require( 'lodash' );
 
 /**
  * Internal dependencies.
  */
 const getIntermediateRepresentation = require( './get-intermediate-representation' );
 
-const getAST = ( source ) =>
-	espree.parse( source, {
-		attachComment: true,
-		loc: true,
-		ecmaVersion: 2018,
-		ecmaFeatures: {
-			jsx: true,
-		},
+const getAST = ( source ) => {
+	return babel.parse( source || '', {
+		plugins: [ 'jsx' ],
```

Comment:
For testing, I'm using this code that includes a variety of new syntax (nullish coalescencing, optional chaining, optional catch binding, and numerical separator):

```js
	try {
		return state?.currentUser;
	} catch {
		return state ?? 1_000;
	}
```

try adding it to the `getCurrentUser` selector within the `packages/core-data/src/selectors.js` file and see how `npm run docs:build` fails due to unrecognizing the numerical separator syntax. The same code works fine with `npm run build`.

To be fully compatible with our current setup, we need to add the [numerical separator proposal](https://babeljs.io/docs/en/next/babel-parser#ecmascript-proposals-https-githubcom-babel-proposals) as a plugin as well:

```suggestion
		plugins: [ 'jsx', 'numericalSeparator' ],
```

The reason is that our `@wordpress/babel-preset-default` config [makes use](https://github.com/WordPress/gutenberg/blob/master/packages/babel-preset-default/index.js#L20) of the [shippedProposals](https://babeljs.io/docs/en/babel-preset-env#shippedproposals) option, which, at the time of writing, also enables the numeric separator syntax ([see](https://github.com/babel/babel/blob/v7.9.5/packages/babel-compat-data/scripts/data/plugin-features.js#L124)).

Reply:
How about we add a comment here that says something along these lines: 

When upgrading babel, we need to make sure this list of plugins are updated to include the shippedProposals that we enable by default in `@wordpress/babel-preset-default` https://github.com/WordPress/gutenberg/blob/master/packages/babel-preset-default/index.js#L20